src/lib/optimizer/strategy/between_composition_rule.cpp

```diff
@@ -0,0 +1,219 @@
+#include "between_composition_rule.hpp"
```

Comment:
This file is missing a whole bunch of comments. There is no high-level explanation of the algorithm and central data structures are not explained. What is a "node scope"? Why can column boundaries be compared by looking at their original column id?

Reply:
is it resolved?

Reply:
Except for the comments below, yes. :)